docs/design/metadata.md

```diff
@@ -0,0 +1,365 @@
+# Metadata formats
+
+## Purpose
+
+This document describes the three metadata formats used for variant
+wheels: `pyproject.toml` tables, distribution metadata fields
```

Comment:
"distribution metadata fields" may be slightly ambiguous. How about saying there's four rather than three formats and explicitly naming `PKG-INFO` and `METADATA`?

Reply:
Hmm, how about "Core Metadata (used in `*.dist-info/METADATA` and `PKG-INFO` files)"?

Reply:
Sounds good, thanks - resolving.